sark/sark_alloc.c

```diff
@@ -56,6 +56,14 @@ void *sark_xalloc(heap_t *heap, uint size, uint tag, uint flag)
     uint entry = (app_id << 8) + tag;
 
     if (tag != 0 && sv->alloc_tag[entry] != NULL) {
+        if (flag & ALLOC_TAG_RETRY) {
```

Comment:
This assumes that a tag will have only one allocation. This is true in your use-case, but it should be made explicit…

Reply:
This has been made explicit in the doc for alloc retry.  Note that this was the previous behaviour too.